plutus-tx-plugin/test/Plugin/Spec.hs

```diff
@@ -82,6 +82,8 @@ primitives = testNested "primitives" [
   --, goldenPlc "blocknum" blocknumPlc
   , goldenPir "bytestring" bytestring
   , goldenEval "bytestringApply" [ getPlc bytestring, unsafeLiftProgram ("hello"::ByteString) ]
+  , goldenEval "sha2_256" [ getPlc sha2, unsafeLiftProgram ("hello" :: ByteString)]
```

Comment:
@effectfully @michaelpj Could you have a look at this test please? It results in `Failure` (see below), I don't know why!

Reply:
Could you please show the output of `sha2 "hello"`?

Reply:
It seems like the output somehow does not fit into a `ByteString` of length 32.

Reply:
Or at least I can't think of a different reason a `Failure` can be returned.

Reply:
yeah looks like that is the reason. The hash has 63 bytes.

```
>Hash.sha2 ("hello" :: BSL.ByteString)
"\159\CAN,\CAN\242\CANM\CAN\186\CAN_\CAN\176\CAN\163\SO\CAN&\CAN\232\CAN;\CAN*\CAN\197\CAN\185\CAN\226\CAN\158\CAN\ESC\SYN\CAN\RS\CAN\\\CAN\US\CAN\167\CANB\CAN^\CANs\EOT\CAN3\CANb\CAN\147\CAN\139\CAN\152\CAN$\255"
>BSL.length (Hash.sha2 ("hello" :: BSL.ByteString))
63
>
```

Reply:
ok. Looks like the size in the spec is wrong. I'll get back to this later today.

Reply:
or rather, my implementation of sha2 and sha3